pkg/install/resource/instance.go

```diff
@@ -0,0 +1,138 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/datawire/dlib/dlog"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+)
+
+// An Instance exposes CRUD operations for a k8s resource such as
+// a Service.
+type Instance interface {
+	Create(context.Context) error
+	Exists(context.Context) (bool, error)
+	Delete(context.Context) error
+	Update(context.Context) error
+}
+
+// scope contains everything that needs to be shared between resources
+// during an operation that spans several resources.
+type scope struct {
+	namespace  string
+	clusterID  string
+	tmSelector map[string]string
+	client     *kates.Client
+	env        *client.Env
+	caPem      []byte
+	crtPem     []byte
+	keyPem     []byte
+}
+
+// The scope is available through the context
+type scopeKey struct{}
+
+func withScope(ctx context.Context, scope *scope) context.Context {
+	return context.WithValue(ctx, &scopeKey{}, scope)
+}
+
+func getScope(ctx context.Context) *scope {
+	if sc, ok := ctx.Value(&scopeKey{}).(*scope); ok {
```

Comment:
Those should just be `scopeKey{}`, not `&scopeKey{}`.

Honestly, I'm shocked it seems to work at all with the `&`.

Reply:
Not intended. I'll remove that ampersand. IIRC, it works but the behavior is undefined in the spec.